apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -0,0 +1,161 @@
+define(function(require) {
+  'use strict';
+
+  var SettingsService = require('modules/settings_service');
+  var ManifestHelper = require('shared/manifest_helper');
+  var AddonManager = require('modules/addon_manager');
+  var Toaster = require('shared/toaster');
+
+  const PREFERRED_ICON_SIZE = 64 * (window.devicePixelRatio || 1);
+
+  /**
+   * @class AddonDetails
+   * @requires module:modules/settings_service
+   * @requires module:shared/manifest_helper
+   * @requires module:modules/addon_manager
+   * @requires module:shared/toaster
+   * @returns {AddonDetails}
+   */
+  function AddonDetails(panel) {
+    this.panel = panel;
+    this.body = panel.querySelector('.addon-details-body');
+    this.header = panel.querySelector('.addon-details-header');
+    this.icon = panel.querySelector('.addon-details-icon');
+    this.description = panel.querySelector('.addon-description-text');
+    this.developer = panel.querySelector('.addon-developer');
+    this.targetsList = panel.querySelector('.addon-targets');
+    this.noTargetsMsg = panel.querySelector('.addon-no-targets');
+    this.toggle = panel.querySelector('.addon-enabled');
+    this.deleteButton = panel.querySelector('.addon-delete');
+  }
+
+  /**
+   * Render details of an app to the UI elements.
+   *
+   * @access public
+   * @memberOf AddonDetails.prototype
+   * @param {DOMApplication} app
+   */
+  AddonDetails.prototype.render = function render(app) {
+    var self = this;
+    var l10n = navigator.mozL10n;
+    this.app = app;
+    var manifest = new ManifestHelper(app.manifest || app.updateManifest);
+
+    // Scroll back to the top
+    this.body.scrollTop = 0;
```

Comment:
This looks more like cleanup works for me, maybe we can directly do this in `onHide()` ?